File: src/brs/http/APITransactionManager.java
package brs.http;

import brs.Account;
import brs.Attachment;
import brs.BurstException;
import org.json.simple.JSONStreamAware;

import javax.servlet.http.HttpServletRequest;

public interface APITransactionManager {

  JSONStreamAware createTransaction(HttpServletRequest req, Account senderAccount, Long recipientId, long amountNQT, Attachment attachment, long minimumFeeNQT) throws BurstException;

}

File: src/main/java/brs/http/APITransactionManager.kt
package brs.http

import brs.Account
import brs.Attachment
import brs.BurstException
import com.google.gson.JsonElement

import javax.servlet.http.HttpServletRequest

interface APITransactionManager {
    @Throws(BurstException::class)
    fun createTransaction(req: HttpServletRequest, senderAccount: Account, recipientId: Long?, amountNQT: Long, attachment: Attachment, minimumFeeNQT: Long): JsonElement
}
